modules/git/git.go

```diff
@@ -251,6 +251,19 @@ func syncGitConfig() (err error) {
 		}
 	}
 
+	// Disable the security check because Gitea runs the git CLI from within the
+	// repository. See https://github.com/go-gitea/gitea/issues/19455 for the full discussion.
+	// safe.directory accept * was introduced in https://github.com/git/git/blob/main/Documentation/RelNotes/2.30.4.txt
+	if CheckGitVersionAtLeast("2.30.4") == nil {
+		if err := configAddNonExist("safe.directory", "*"); err != nil {
+			return err
+		}
+	} else {
+		if err := configSet("safe.directory", setting.RepoRootPath); err != nil {
+			return err
+		}
+	}
+
```

Comment:
```suggestion
	if err := configAddNonExist("safe.directory", "*"); err != nil {
		return err
	}
```

(according to that Git version problem is very complex, and it won't cause problems for Git with lower version)